### ISSUE TITLE
Knife vault refresh fails: `Error: Missing read permission`

### ISSUE THREAD

**Issue by xakraz** [chef/chef-vault#225]
Hello,

I am facing an issue with `knife vault refresh` and have this blurry message.

```
Starting run 1474473841 (Wed Sep 21 16:04:01 UTC 2016)
--> Knife using Chef-vault Gem:
2.9.0
...

knife vault refresh shared_secrets VAULT_ITEM

Refreshing shared_secrets/VAULT_ITEM
ERROR: You authenticated successfully to https://127.0.0.1/organizations/ORG_NAME as USERX but you are not authorized for this action
Response:  missing read permission
```

We are using:
- chef-server `12.4.1`
- A cron based job that does a `knife vault refresh` periodically to update the clients list for each vault items according to the `search_query`

Everything was working well and suddenly, with no obvious reasons and changes, we have this error.

I have already read the issue #171 but it does not seem to be related.

More info:
### Workstation

```
$ ruby --version
ruby 2.1.8p440 (2015-12-16 revision 53160)
```

```
chef --version
Chef Development Kit Version: 0.17.17
chef-client version: 12.13.37
delivery version: master (f68e5c5804cd7d8a76c69b926fbb261e1070751b)
berks version: 4.3.5
kitchen version: 1.11.1
```
### My Chef repo

```
git:(master)   $ grep chef Gemfile.lock

    chef (12.13.37)
    chef-config (12.13.37)
    chef-dk (0.17.17)
    chef-provisioning (1.9.1)
    chef-vault (2.9.0)
    chef-zero (4.9.0)
    cheffish (3.0.1)
    chefspec (5.1.1)
    librarian-chef (0.0.4)
```
### Chef-server

```
[chef02 ~] $ yum list installed | grep -i chef
chef.x86_64                    12.8.1-1.el6              installed
chef-server-core.x86_64        12.4.1-1.el6              installed
```

**Comment by xakraz** [comment]
Hello ! 

We have managed to identify and work around the issue:

## TL;DR

It turned out that our chef-server had a "corrupted" client record.

```
knife client show 'FAULTY_CLIENT'

ERROR: You authenticated successfully to https://CHEF_SERVER_URL/organizations/ORG_NAME as USERX but you are not authorized for this action
Response:  missing read permission
```

Fix:
```
knife client delete 'FAULTY_CLIENT'
knife client create 'FAULTY_CLIENT' -p public_key.pem

```

## Details:

We identified the issue while looking at the chef-server embedded nginx access logs:


```
// VAULT encrypted data bag
127.0.0.1 - - [20/Sep/2016:13:08:20 +0000]  "GET /organizations/ORG_NAME/data/shared_secrets/VAULT_ITEM_keys HTTP/1.1" 200 "0.020" 5241 "-" "Chef Knife/12.8.1 (ruby-2.1.6-p336; ohai-8.11.1; x86_64-linux; +https://chef.io)" "127.0.0.1:8000" "200" "0.019" "12.8.1" "algorithm=sha1;version=1.1;" "USERx" "2016-09-20T13:08:20Z" "2jmj7l5rSw0yVb/vlWAYkK/YBwk=" 1054
127.0.0.1 - - [20/Sep/2016:13:08:20 +0000]  "GET /organizations/ORG_NAME/data/shared_secrets/VAULT_ITEM HTTP/1.1" 200 "0.019" 202 "-" "Chef Knife/12.8.1 (ruby-2.1.6-p336; ohai-8.11.1; x86_64-linux; +https://chef.io)" "127.0.0.1:8000" "200" "0.019" "12.8.1" "algorithm=sha1;version=1.1;" "USERx" "2016-09-20T13:08:20Z" "2jmj7l5rSw0yVb/vlWAYkK/YBwk=" 1049

// Search Query
127.0.0.1 - - [20/Sep/2016:13:08:24 +0000]  "GET /organizations/ORG_NAME/search/node?q=*:*&sort=X_CHEF_id_CHEF_X%20asc&start=0 HTTP/1.1" 200 "4.219" 53686456 "-" "Chef Knife/12.8.1 (ruby-2.1.6-p336; ohai-8.11.1; x86_64-linux; +https://chef.io)" "127.0.0.1:8000" "200" "4.218" "12.8.1" "algorithm=sha1;version=1.1;" "USERx" "2016-09-20T13:08:20Z" "2jmj7l5rSw0yVb/vlWAYkK/YBwk=" 1069

// Get clients info
127.0.0.1 - - [20/Sep/2016:13:09:45 +0000]  "GET /organizations/ORG_NAME/clients/CLIENT1 HTTP/1.1" 200 "0.022" 525 "-" "Chef Knife/12.8.1 (ruby-2.1.6-p336; ohai-8.11.1; x86_64-linux; +https://chef.io)" "127.0.0.1:8000" "200" "0.018" "12.8.1" "algorithm=sha1;version=1.1;" "USERx" "2016-09-20T13:09:45Z" "2jmj7l5rSw0yVb/vlWAYkK/YBwk=" 1057
127.0.0.1 - - [20/Sep/2016:13:09:45 +0000]  "GET /organizations/ORG_NAME/clients/CLIENT2 HTTP/1.1" 200 "0.017" 527 "-" "Chef Knife/12.8.1 (ruby-2.1.6-p336; ohai-8.11.1; x86_64-linux; +https://chef.io)" "127.0.0.1:8000" "200" "0.016" "12.8.1" "algorithm=sha1;version=1.1;" "USERx" "2016-09-20T13:09:45Z" "2jmj7l5rSw0yVb/vlWAYkK/YBwk=" 1057

...

// FAULTY CLIENT
127.0.0.1 - - [20/Sep/2016:13:10:45 +0000]  "GET /organizations/ORG_NAME/clients/FAULTY_CLIENT HTTP/1.1" 403 "0.023" 37 "-" "Chef Knife/12.8.1 (ruby-2.1.6-p336; ohai-8.11.1; x86_64-linux; +https://chef.io)" "127.0.0.1:8000" "403" "0.023" "12.8.1" "algorithm=sha1;version=1.1;" "USERx" "2016-09-20T13:10:45Z" "2jmj7l5rSw0yVb/vlWAYkK/YBwk=" 1057

// New 'knife vault refresh' command (loop from Shell scrip)
127.0.0.1 - - [20/Sep/2016:13:10:47 +0000]  "GET /organizations/criteo/data/shared_secrets/NEX_VAULT_ITEM_keys HTTP/1.1" 200 "0.026" 5252 "-" "Chef Knife/12.8.1 (ruby-2.1.6-p336; ohai-8.11.1; x86_64-linux; +https://chef.io)" "127.0.0.1:8000" "200" "0.025" "12.8.1" "algorithm=sha1;version=1.1;" "a_boten" "2016-09-20T13:10:47Z" "2jmj7l5rSw0yVb/vlWAYkK/YBwk=" 1054
```

So the current behaviour is that : 
* `knife vault refresh` fails when it received a 403 HTTP response for a specific client 
* And the vault is not updated 
* We don't have a good error message...

**Comment by kamaradclimber** [comment]
symptom fixed by #231, the rest (failing due to 403) looks an expected behavior
